caddyhttp/staticfiles/fileserver.go

```diff
@@ -121,13 +125,41 @@ func (fs FileServer) serveFile(w http.ResponseWriter, r *http.Request, name stri
 		return http.StatusNotFound, nil
 	}
 
+	filename := d.Name()
+
+	for _, encoding := range staticEncodingPriority {
+		if strings.Contains(r.Header.Get("Accept-Encoding"), encoding) {
+			gf, err := fs.Root.Open(location + staticEncoding[encoding])
+
+			if err == nil {
+				stat, err := gf.Stat()
+
+				if err == nil {
+					// Close previous file - release fd
+					f.Close()
```

Comment:
I've checked and in case that file was previously closed ```Close()``` will return error (```ErrInvalid  - invalid argument```). As we don't check ```Close()``` return error calling it is no-op.